x/observer/keeper/msg_server_add_block_header.go

```diff
@@ -35,6 +34,38 @@ func (k msgServer) AddBlockHeader(goCtx context.Context, msg *types.MsgAddBlockH
 		return nil, cosmoserrors.Wrapf(types.ErrBlockHeaderVerficationDisabled, "proof verification not enabled for evm ,chain id: %d", msg.ChainId)
 	}
 
+	_, found = k.GetBlockHeader(ctx, msg.BlockHash)
+	if found {
+		return nil, cosmoserrors.Wrap(types.ErrBlockAlreadyExist, "cannot add block header")
```

Comment:
```suggestion
		return nil, cosmoserrors.Wrap(types.ErrBlockAlreadyExist, msg.BlockHash)
```
I think we should use error wrapping to make the error more explicit with the context of the call